mlflow/spark.py

```diff
@@ -666,14 +672,21 @@ def autolog():
                 (7, "apache hadoop")], ["id", "text"])
         import tempfile
         tempdir = tempfile.mkdtemp()
-        df.write.format("csv").save(tempdir)
+        df.write.csv(os.path.join(tempdir, "my-data-path"), header=True)
         # Enable Spark datasource autologging.
         mlflow.spark.autolog()
-        loaded_df = spark.read.format("csv").load(tempdir)
-        # Call collect() to trigger a read of the Spark datasource. Datasource info
-        # (path and format)is automatically logged to an MLflow run.
-        loaded_df.collect()
-        shutil.rmtree(tempdir) # clean up tempdir
+        loaded_df = spark.read.csv(os.path.join(tempdir, "my-data-path"),
+                        header=True, inferSchema=True)
+        # Call toPandas() to trigger a read of the Spark datasource. Datasource info
+        # (path and format) is logged to the current active run, or the
+        # next-created MLflow run if no run is currently active
+        with mlflow.start_run() as active_run:
+            pandas_df = loaded_df.toPandas()
+            text = pandas_df["text"]
```

Comment:
We should be able to delete the lines after the `toPandas` call :), sorry I left those in when sharing a snipper with you

Reply:
hehe true i just tested the code and it worked so i didnt read it :^)